insights/util/subproc.py

```diff
@@ -79,26 +79,35 @@ def call(cmd, timeout=None, signum=signal.SIGKILL, shell=False, stdout=subproces
 
     output = None
     rc = 0
+    clist = False
     try:
-        cmd = cmd.encode('utf-8', 'replace')
-        if timeout is not None and sys.platform != "darwin":
-            cmd = "timeout -s {0} {1} {2}".format(signum, timeout, cmd)
+        if type(cmd) is list:
+            clist = True
+            command = cmd
+            cmd = []
+            for cl in command:
+                cmd += [[c.encode('utf-8', 'replace') for c in cl]]
+            if timeout is not None and sys.platform != "darwin":
+                cmd[0] = ['timeout', '-s', str(signum), str(timeout)] + cmd[0]
+        else:
+            cmd = cmd.encode('utf-8', 'replace')
+            if timeout is not None and sys.platform != "darwin":
+                cmd = "timeout -s {0} {1} {2}".format(signum, timeout, cmd)
 
         log.debug(cmd)
 
         if not shell:
-            spltcmd = cmd.split("|")
-            if len(spltcmd) > 1:
-                cmd = shlex.split(spltcmd[0])
-                cout = Popen(cmd, stdout=stdout)
-                del spltcmd[0]
-
-                for next in spltcmd:
-                    nxt = shlex.split(next)
-                    cout = Popen(nxt, stdout=stdout, stderr=stderr, stdin=cout.stdout)
+            if not clist:
+                spltcmd = cmd.split("|")
```

Comment:
I'd take the split on pipe out since it'll break if the regular string has a pipe in a quoted string or something. Calls can just pass a list of lists for piping so we don't have to worry about parsing.

Reply:
I left the split in there for backward compat. The original version of subproc had the split in it 'if not shell' so I figured there may be some other things using subproc passing a string with a pipe that count on the split. Just didn't want to break something.

Reply:
Removed split on pipe from subproc.